src/mod_auth_cas.c

```diff
@@ -1741,15 +1741,22 @@ apr_byte_t isValidCASCookie(request_rec *r, cas_cfg *c, char *cookie, char **use
 
 size_t cas_curl_write(const void *ptr, size_t size, size_t nmemb, void *stream)
 {
+	size_t realsize = size * nmemb;
```

Comment:
FYI: it seems possible that this multiplication could overflow in some circumstances, in which case the memcpy() below would write more than has been allocated.  Simlarly, curlBuffer->written + realsize + 1 also risks overflow.

Even if you remove a hard static limit, I would still suggest having a reasonable cap (i.e. probably no response should ever exceed say... 4MB? 16MB?) to guard against a malicious CAS server, or people testing over http://.

Reply:
CURLOPT_WRITEFUNCTION notes that CURL_MAX_WRITE_SIZE bytes can be sent to this function, the default being 16k.  

Assuming a 64 bit size_t, 9 exabytes of data would have to come through in a request for this to overflow. That seemed to be so much in the realm of preposterousness that I did not worry about it.

I'm not opposed to capping at some large amount, but I was trying to remove those limitations here.

Reply:
CURLOPT_WRITEFUNCTION also states that _size_ is always 1, so this shouldn't be a problem in practice.

Even so, I've added a commit to bring back CAS_MAX_RESPONSE_SIZE with a large (2GiB) value to guard against overflows and hopefully ensure that users will never have to tweak this value.

Do you find that acceptable?

Reply:
This seems like a good compromise between very desirable dynamic behavior and static limits. While we might have further discussion about the default static limit value (2G), I'd like to take a win here and have that discussion at a later date if there's some further evidence it's needed.